app/src/main/java/ar/com/wolox/android/example/ui/example/ExamplePresenter.kt

```diff
@@ -4,19 +4,48 @@ import ar.com.wolox.android.example.utils.UserSession
 import ar.com.wolox.wolmo.core.presenter.BasePresenter
 import javax.inject.Inject
 
+fun flattenSize(m: MutableMap<String, MutableList<String>>): Int {
+    var acc = 0
+    m.forEach { (s, mutableList) -> acc += mutableList.size }
+    return acc
+}
+
 class ExamplePresenter @Inject constructor(private val userSession: UserSession) : BasePresenter<ExampleView>() {
 
-    fun onLoginButtonClicked(user: String, color: String) {
-        userSession.username = user
-        view?.goToViewPager(color)
-    }
+    fun onLoginButtonClicked(email: String, password: String) {
+        var ErrorList: MutableMap<String, MutableList<String>> = mutableMapOf()
+
+        // TODO: How to avoid these lines ? I mean, how to create ErrorList[...] while inserting element in its list
+        ErrorList["email"] = mutableListOf()
+        ErrorList["password"] = mutableListOf()
 
-    fun onEmptyFieldsDetected() = null
+        // TODO: How can I make this lines "more" general ?
+        if (email.trim() == EMPTY_STRING) ErrorList["email"]?.add(ERROR_INPUT)
+        if (password.trim() == EMPTY_STRING) ErrorList["password"]?.add(ERROR_INPUT)
+        if (!android.util.Patterns.EMAIL_ADDRESS.matcher(email.trim()).matches()) ErrorList["email"]?.add(ERROR_BAD_EMAIL)
 
+        if (flattenSize(ErrorList) == 0) {
+            println("ENTRE")
```

Comment:
I forgot to delete this line...

Reply:
lol xd no problem, delete it in the next commit